### DISCUSSION TITLE
Cannot import resources into `eks-cluster` module: Invalid provider configuration

### DISCUSSION THREAD

**Discussion by jeffreymlewis** [gruntwork-io/knowledge-base#747] [Help]
Hello,

I'm using [service catalog 'eks-cluster' v0.104.14](https://github.com/gruntwork-io/terraform-aws-service-catalog/tree/v0.104.14/modules/services/eks-cluster) to deploy an EKS 1.26 cluster. My AWS credentials expired mid-way through the initial `terragrunt apply` and terraform couldn't save the remote state.

I cleared the lockfile and tried running `terragrunt apply` again, which failed (expectantly) with errors like

```
│ Error: creating Security Group (dev00-ue1-app-eks-03): InvalidGroup.Duplicate: The security group 'dev00-ue1-app-eks-03' already exists for VPC 'vpc-xxxxxxxxxxxxxxx'
```

I then tried importing this resource into the remote state, and it failed with.

```
terragrunt import 'module.eks_cluster.aws_security_group.eks' sg-xxxxxxxxxxxxxxx
...
...
│ Error: Invalid provider configuration
│ 
│   on /Users/..../infrastructure-live/dev/us-east-1/dev/services/eks-cluster-03/eks-cluster/.terragrunt-cache/nDw0NWFxYqtZsc8AAJmRCnoMYrw/fCw8UgbtyEKZwmAHcT_oAWkqDzs/modules/services/eks-cluster/main.tf line 41:
│   41: provider "kubernetes" {
│ 
│ The configuration for
│ provider["registry.terraform.io/hashicorp/kubernetes"] depends on values
│ that cannot be determined until apply.
╵

Releasing state lock. This may take a few moments...
```

I expect to be able to import resources into the `eks-cluster` module. Not sure if I have something mis-configured in terragrunt, or if this is a bug.

Incidentally, running `terragrunt destroy` also results in an error. (Probably a different bug, deserving it's own ticket.)

```
terragrunt destroy
...
...
Planning failed. Terraform encountered an error while generating this plan.

│ Error: Unsupported attribute
│ 
│   on ../eks-workers/managed_node_groups.tf line 266, in locals:
│  266:   managed_node_group_asg_names = flatten([for name, asg_names in module.managed_node_groups.eks_worker_asg_names : asg_names])
│     ├────────────────
│     │ module.managed_node_groups is object with 2 attributes
│ 
│ This object does not have an attribute named "eks_worker_asg_names".
╵
Releasing state lock. This may take a few moments...
```

**Comment by pras111gw** [comment]
Hi @jeffreymlewis In this situation, easiest path out is probably to destroy and recreate the cluster. Would you mind trying that?

**Reply by jeffreymlewis** [reply]
Running `terragrunt destroy` produced the error shown above (... object does not have an attribute named "eks_worker_asg_names") which seems like a bug.

I fixed this by manually deleting resources from AWS. 🙁 I think we can close this.
